third_party/2and3/click/types.pyi

```diff
@@ -26,9 +28,17 @@ class FloatParamType(ParamType):
     def convert(self, value: str, param: Optional[Parameter], ctx: Optional[Context]) -> float: ...
 
 class FloatRange(FloatParamType):
+    min: Optional[float]
+    max: Optional[float]
+    clamp: bool
```

Comment:
Not a show stopper, but I would prefer these to be implemented on `_NumberRangeBase`, like in the implementation.

Reply:
I can do this, but have not initially done in accordance with the typshed contributing guide lines that state
> Note: in its current implementation, typeshed cannot contain stubs for multiple versions of the same third-party library. Prefer to generate stubs for the latest version released on PyPI at the time of your stubbing.

As of now the latest release for click is [7.1.2](https://pypi.org/project/click/7.1.2/) for which the base class `_NumberRangeBase` does not exist yet. See https://github.com/pallets/click/blob/1784558ed7c75c65764d2a434bd9cbb206ca939d/src/click/types.py

This will be introduced with the next version 8.0 for which for now exists only a pre-release [8.0.0a1](https://pypi.org/project/click/8.0.0a1/). 

So if you prefer I can adapt this PR to be based on the current state of the pre-release which in addition to your suggestion  would also mean adding the new attributes/init args `min_open` and `max_open` to the two number range types. There might also be some other changes specifically in the types file.

Reply:
Makes sense, I didn't notice that the current version does not have the base class.